### PR TITLE
modify run-pytests to check version before test ray

### PR DESCRIPTION
**Pull request by respecteverything**

### CODE REVIEW

**Comment by respecteverything** [comment]
PR-Validation Successful : http://10.239.47.211:18888/job/ZOO-PR-Validation/2477/

File: pyzoo/zoo/examples/run-example-tests-pip.sh [hunk 1]
```diff
@@ -411,6 +411,7 @@ if [[ $py_version == *"Python 3.5"* || $py_version == *"Python 3.6"* ]]; then
         fi
         echo "Install gym"
         pip install gym
+        pip install gym[atari]
```
Comment (zhichao-li) [inline pyzoo/zoo/examples/run-example-tests-pip.sh]: This should be moved to Jenkins image before merging the PR.

File: pyzoo/zoo/examples/ray/.keep [hunk 2]
```diff
@@ -0,0 +1 @@
+first pr
```
Comment (zhichao-li) [inline pyzoo/zoo/examples/ray/.keep]: remove this file

File: pyzoo/zoo/examples/run-example-tests-pip.sh [hunk 3]
```diff
@@ -375,6 +376,75 @@ now=$(date "+%s")
...
+    return $((now-start))
+}
+
+spark_version="$($SPARK_HOME/bin/pyspark --version 2>&1)"
+temp="${spark_version% /_/*}"
+spark_version="${temp##*\\}"
+py_version="$(python -V 2>&1)"
+
+if [[ $py_version == *"Python 3.5"* || $py_version == *"Python 3.6"* ]]; then
+    if [[ $spark_version == *"version 2.4.3"* || $spark_version == *"version 2.4.0"* ]]; then
```
Comment (zhichao-li) [inline pyzoo/zoo/examples/run-example-tests-pip.sh]: It's a nice try, but I think we should let Jenkins make the decision as to what it does for the other job.

File: pyzoo/zoo/examples/run-example-tests-pip-ray.sh [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+#if image exist this two dependency, remove below
+echo "Install gym and atari"
+pip install gym
+pip install gym[atari]
+
+#start execute
+echo "Start pong example"
+start=$(date "+%s")
+export SPARK_DRIVER_MEMORY=4g
+python ${ANALYTICS_ZOO_ROOT}/pyzoo/zoo/examples/ray/rl_pong/rl_pong.py \
```
Comment (zhichao-li) [inline pyzoo/zoo/examples/run-example-tests-pip-ray.sh]: Extract this into function. you can refer to : 
https://github.com/intel-analytics/analytics-zoo/pull/1420/files#diff-fe78914e3f2e8b9fe1752067c7539f4b
```
my_function () {  
  elapse_time=$((now-start))
} 
 
time9=${elapse_time}
```

File: pyzoo/zoo/examples/run-example-tests-pip.sh [hunk 5]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env bash
 
+
```
Comment (zhichao-li) [inline pyzoo/zoo/examples/run-example-tests-pip.sh]: revert these unnecessary changes.

File: pyzoo/dev/run-pytests [hunk 6]
```diff
@@ -24,12 +24,45 @@ export PYSPARK_PYTHON=python
...
+        echo "current spark version is not support ray."
+        echo "Skipping ray. Please check your spark version or check SPARK_HOME."
+        python -m pytest -v --doctest-modules ../zoo \
+            --ignore=../zoo/pipeline/api/keras2 \
+            --ignore=../zoo/tfpark/text \
+            --ignore=../zoo/examples \
+            --ignore=../zoo/ray/
+    fi
+else
+    echo "current python version is not support ray."
```
Comment (zhichao-li) [inline pyzoo/dev/run-pytests]: I think we can have a dedicated file for the ray test. e.g `run-ray-pytests.py`
```
    python -m pytest -v ../test/zoo/ray/
       --ignore=../test/zoo/ray/integration/
    exit_status_3=$?
    if [ $exit_status_3 -ne 0 ];
    then
        exit $exit_status_3
    fi
```

Comment (respecteverything) [reply pyzoo/dev/run-pytests]: put check_version to run_tests and move this pytest_ray code  into run_pytest_ray file?

**Comment by respecteverything** [comment]
jenkins python example test success: http://10.239.47.211:18888/job/ZOO-PR-Python-ExampleTests/2313/

**Comment by respecteverything** [comment]
PR-Validation Success : http://10.239.47.211:18888/job/ZOO-PR-Validation/2537/
